pkg/cmdutil/userio/single_select.go

```diff
@@ -79,15 +92,23 @@ func (m singleSelectModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
 		} else {
 			m.model.SetHeight(msg.Height)
 		}
-		return m, nil
+		newListModel, cmd := m.model.Update(msg)
+		m.model = newListModel
+		return m, cmd
 
 	case tea.KeyMsg:
 		switch msg.Type {
 		case tea.KeyCtrlC, tea.KeyEsc:
 			m.choice = nil
 			m.err = ErrInterrupted
-			m.quitting = true
-			return m, tea.Quit
+			if m.embedded {
+				return m, func() tea.Msg {
+					return InputCompleted{model: m}
```

Comment:
Why is it only necessary for single select? Can we somehow work around it, so the component doesn't care, if it's embedded or not?

Reply:
I've not changed how any of the other commands which use other components work yet. I'd prefer it if the component didn't have to know, but we can't allow the `tea.Quit` to be sent, or it gets picked up by the parent and exits the program too early. Let me know if you can find a nicer way, we might be able to intercept the tea.Quit, but it's not too pretty either: https://github.com/charmbracelet/bubbletea/issues/472

Reply:
I think this handler should solve the problem, since all this complexity will be one place.
Plus, if I understand it correctly, you'll just have to check `wizardModel.quitting` variable to make a decision

Reply:
I've changed over to using WithFilter to capture the quit events